compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -64,7 +64,8 @@ object DesugarEnums {
   }
 
   /** A type tree referring to `enumClass` */
-  def enumClassRef(implicit ctx: Context) = TypeTree(enumClass.typeRef)
+  def enumClassRef(implicit ctx: Context) =
+    if (enumClass.exists) TypeTree(enumClass.typeRef) else TypeTree()
```

Comment:
By retruning an empty TypeTree here, calls to `enumClassRef` are made safe even from other classes.